ice-rest-catalog/src/main/java/com/altinity/ice/rest/catalog/internal/maintenance/OrphanFileScanner.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2025 Altinity Inc and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.altinity.ice.rest.catalog.internal.maintenance;
+
+import com.altinity.ice.cli.internal.s3.S3;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.io.FileIO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+
+public class OrphanFileScanner {
+  private static final Logger LOG = LoggerFactory.getLogger(OrphanFileScanner.class);
+  private final Table table;
+
+  public OrphanFileScanner(Table table) {
+    this.table = table;
+  }
+
+  private Set<String> getAllKnownFiles() {
+    Set<String> knownFiles = new HashSet<>();
+
+    for (Snapshot snapshot : table.snapshots()) {
+      // Manifest list file
+      if (snapshot.manifestListLocation() != null) {
+        knownFiles.add(snapshot.manifestListLocation());
+      }
+
+      // Manifest files
+      FileIO io = table.io();
+      for (ManifestFile manifest : snapshot.dataManifests(io)) {
+        knownFiles.add(manifest.path());
+      }
+    }
+
+    return knownFiles;
```

Comment:
The result looks incomplete. Inserting a single file into empty table yields
```
./metadata/00000-f62c927f-3965-42e9-8f3a-57809ca49f55.metadata.json
./metadata/4b00c8c2-6d25-4ba5-bf27-989fbdb9daa1-m0.avro
./metadata/snap-7186194791324723249-1-4b00c8c2-6d25-4ba5-bf27-989fbdb9daa1.avro
./metadata/00001-07fe7fa7-8fe1-4666-bd7b-7c88b9c861fe.metadata.json
./data/pickup_datetime_month=2009-01/1750715797525-2d861d7a4e5196bf04e5c3be506b0152f4170cf4e941424c8245b46903c3e252-part.parquet
```
i.e. we're going to nuke a bunch of critical files and break the catalog...